python/lsst/ts/ledprojector/ledprojector_csc.py

```diff
@@ -0,0 +1,297 @@
+# along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+__all__ = ["LEDProjectorCsc", "run_ledprojector"]
+
+import asyncio
+import types
+from typing import Any, List, Union
+
+from lsst.ts import salobj
+from lsst.ts.xml.enums.LEDProjector import LEDBasicState
+
+from . import __version__
+from .config_schema import CONFIG_SCHEMA
+from .led_controller import COMMUNICATION_TIMEOUT, LEDController
+
+
+class LEDProjectorCsc(salobj.ConfigurableCsc):
+    """LED Projector controller
+
+    Parameters
+    ----------
+    config_dir : `None` or `str`, optional
+        Directory of configuration files, or None for the standard
+        configuration directory (obtained from `_get_default_config_dir`).
+        This is provided for unit testing.
+    initial_state : `State` or `int`, optional
+        The initial state of the CSC. This is provided for unit testing,
+        as real CSCs should start up in `State.STANDBY`, the default.
+    override : `str`, optional
+        Configuration override file to apply if ``initial_state`` is
+        `State.DISABLED` or `State.ENABLED`.
+    simulation_mode : `bool`, optional
+        Simulation mode; one of:
+
+        * False for normal operation
+        * True for simulation
+
+    Attributes
+    ----------
+    led_controller : `LEDController`
+        The controller representing the labjack controller
+        which is controlling the leds.
+    """
+
+    version = __version__
+    valid_simulation_modes = (0, 1)
+
+    def __init__(
+        self,
+        config_dir: None | str = None,
+        initial_state: salobj.State = salobj.State.STANDBY,
+        override: str = "",
+        simulation_mode: bool = False,
+    ) -> None:
+        self.led_controller = None
+        self.should_be_connected = False
+
+        self.config = None
+
+        super().__init__(
+            name="LEDProjector",
+            index=None,
+            config_schema=CONFIG_SCHEMA,
+            config_dir=config_dir,
+            initial_state=initial_state,
+            override=override,
+            simulation_mode=simulation_mode,
+        )
+
+    @staticmethod
+    def get_config_pkg() -> str:
+        return "ts_config_mtcalsys"
+
+    @property
+    def controller_connected(self) -> bool:
+        """Return True if the LabJack is connected"""
+        return self.led_controller is not None and self.led_controller.connected
+
+    async def configure(self, config: Any) -> None:
+        """Configure the CSC"""
+        self.config = config
+
+    async def handle_summary_state(self) -> None:
+        """Override of the handle_summary_state function to
+        create led controller object when enabled
+
+        Raises
+        ------
+        ValueError
+            If config is None
+            If Led Controller should be connected but isn't
+        """
+        self.log.info(f"handle_summary_state {salobj.State(self.summary_state).name}")
+        if self.disabled_or_enabled:
```

Comment:
I think you can merge some of the if statements to make it more compact. 

Something like this I think will have the same effect.

```
if self.disabled_or_enabled and self.led_controller is None:
    if self.should_be_connected:
        await self.fault(
            code=LEDBasicState.Error,
            report="led controller should be connected but isn't.",
        )
        raise RuntimeError("LED Controller should be connected but isn't")
    elif self.config is None:
                        raise RuntimeError(
                            "Tried to create LEDController without a configuration. "
                            "This is most likely a bug with the control sequence causing "
                            "the configuration step to be skipped. Try sending the CSC back "
                            "to STANDBY or OFFLINE and restarting it. You should report this issue."
                        )
    else:
        self.led_controller = LEDController(
                        config=self.config,
                        log=self.log,
                        simulate=self.simulation_mode,
                    )

    await self.connect_led()
else:
    await self.disconnect_led()
```